ui/src/components/UploadStory.tsx

```diff
@@ -0,0 +1,855 @@
+        author: {
+          ID: 0,
+          CreatedAt: "",
+          UpdatedAt: "",
+          DeletedAt: {
+            Time: "",
+            Valid: true,
+          },
+          firstName: formInput.author_last_name as string,
+          currentCity: formInput.current_city as string,
+          bio: formInput.bio as string,
+        } as Author,
+        tags: [],
+        is_visible: true,
+        latitude: 0,
+        longitude: 0,
+        ID: 0,
+        CreatedAt: "",
+        UpdatedAt: "",
+        DeletedAt: {
+          Time: "",
+          Valid: true,
+        },
+      };
+      return storyFromData;
+    }
+    return undefined;
+  }, [isDrawerOpen, formInput]);
+
+  const hasAllRequiredFields = React.useMemo(() => {
+    return (
+      formInput.image &&
+      formInput.title &&
+      formInput.content &&
+      formInput.summary &&
+      formInput.author_first_name &&
+      formInput.author_last_name &&
+      formInput.year &&
+      formInput.current_city &&
+      formInput.author_country
+    );
+  }, [formInput]);
+
+  const setImage = (files: File[]) => {
+    setFormInput({
+      image: files[0],
+    });
+  };
+
+  const handleKeyDown = (event) => {
+    switch (event.key) {
+      case ",":
+      case " ": {
+        event.preventDefault();
+        event.stopPropagation();
+        if (event.target.value.length > 0) {
+          setTagArrayValues([...tagArray, event.target.value]);
+        }
+        break;
+      }
+      default:
+    }
+  };
+
+  const setErrorUploadingState = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    setErrorOpen(false);
+  };
+
+  const handleChange = (
+    event?: React.ChangeEvent<{ name?: string; value: unknown }>
+  ) => {
+    const name = event.target.name as keyof typeof formInput;
+    setFormInput({
+      [name]: event.target.value,
+    });
+  };
+
+  const handleDialogClose = () => {
+    setDialogOpenState(false);
+  };
+
+  const handleDialogAgree = () => {
+    handleDialogClose();
+  };
+
+  const handleDialogDisagree = () => {
+    handleDialogClose();
+  };
+
+  const storySubmitDialog = (result) => {
+    setLoading(false);
+    setDisabled(false);
+    const resultMessage = JSON.parse(result).message;
+    if (
+      resultMessage === "Story Added Successfully" ||
+      resultMessage === "Story Updated successfully"
+    ) {
+      //TODO: Fix redirect (currently does nothing)
+      console.log("HERE");
+      return <Redirect to="/admin" />;
+    } else {
+      setErrorOpen(true);
+    }
+  };
+
+  const addCountryButtonPressed = (autocompleteAuthorCountry: string) => {
+    setFormInput({
+      ["author_country"]: autocompleteAuthorCountry,
+    });
+    setAddedCountry(autocompleteAuthorCountry);
+    setAuthorCountry(autocompleteAuthorCountry);
+    countries.push(autocompleteAuthorCountry);
+  };
+
+  const addNewCountry = ({ children, ...other }) => (
+    <AddCountryPaper {...other}>
+      {countries.filter(
```

Comment:
Can we use strict equality/inequality checks in these comparators please

Reply:
Just wondering, what is the benefit of that in this case considering we know that autocompleteAuthorCountry is a string? Is it just best practice?

Reply:
Yeah for readability if someone came into our codebase and didn't know they were the same type and also for consistency sake since we strict equality check everywhere else.